feature/experimental/bgp/otg_tests/bgp_always_compare_med/bgp_always_compare_med_test.go

```diff
@@ -487,12 +494,27 @@ func verifySetMed(t *testing.T, otg *otg.OTG, config gosnappi.Config, wantMEDVal
 	}
 
 	// compare Med val with expected for each of the recieved routes.
-	for _, prefix := range bgpPrefixes {
-		if prefix.GetMultiExitDiscriminator() != wantMEDValue {
-			t.Errorf("Received Prefix Med %d Expected Med %d for Prefix %v", prefix.GetMultiExitDiscriminator(), wantMEDValue, prefix.GetAddress())
+	wantMED := []uint32{}
+	for i := 0; i < routeCount; i++ {
+		wantMED = append(wantMED, uint32(wantMEDValue))
+	}
+
+checkMEDLoop:
+	for repeat := 10; repeat > 0; repeat-- {
+		gotMED := gnmi.GetAll(t, otg, gnmi.OTG().BgpPeer(ateSrc.Name+".BGP4.peer").UnicastIpv4PrefixAny().MultiExitDiscriminator().State())
+		diff := cmp.Diff(wantMED, gotMED)
+		switch {
+		case diff == "":
+			t.Logf("MED values are as expected")
+			break checkMEDLoop
+		case diff != "" && repeat > 0:
+			t.Logf("MED values not as expected , wait for 10 sec before retry. want %v , got %v", wantMED, gotMED)
+			time.Sleep(10 * time.Second)
```

Comment:
Please gnmi.Watch instead of time.sleep
https://github.com/openconfig/ondatra/blob/b07a95699a4eed8aa606762baadc0a96074c4394/gnmi/gnmi.go#L42-L86